fastparquet/test/test_util.py

```diff
@@ -1,28 +1,50 @@
 import pytest
 
-from fastparquet.util import analyse_paths, get_file_scheme, val_to_num
+from fastparquet.util import analyse_paths, get_file_scheme, val_to_num, join_path
 
 
 def test_analyse_paths():
     file_list = ['a', 'b']
-    base, out = analyse_paths(file_list, '/')
+    base, out = analyse_paths(file_list)
     assert (base, out) == ('', ['a', 'b'])
 
     file_list = ['c/a', 'c/b']
-    base, out = analyse_paths(file_list, '/')
+    base, out = analyse_paths(file_list)
     assert (base, out) == ('c', ['a', 'b'])
 
     file_list = ['c/d/a', 'c/d/b']
-    base, out = analyse_paths(file_list, '/')
+    base, out = analyse_paths(file_list)
     assert (base, out) == ('c/d', ['a', 'b'])
 
+    file_list = ['/c/d/a', '/c/d/b']
+    base, out = analyse_paths(file_list)
+    assert (base, out) == ('/c/d', ['a', 'b'])
+
     file_list = ['c/cat=1/a', 'c/cat=2/b', 'c/cat=1/c']
-    base, out = analyse_paths(file_list, '/')
+    base, out = analyse_paths(file_list)
     assert (base, out) == ('c', ['cat=1/a', 'cat=2/b', 'cat=1/c'])
 
-    file_list = ['c\\cat=1\\a', 'c\\cat=2\\b', 'c\\cat=1\\c']
-    base, out = analyse_paths(file_list, '\\')
-    assert (base, out) == ('c', ['cat=1\\a', 'cat=2\\b', 'cat=1\\c'])
+
+def test_empty():
+    assert join_path("test", ""), "test"
+
+
+def test_parents():
+    assert join_path("test", "../../..") == "../.."
+
+    with pytest.raises(Exception):
+        join_path("/test", "../../..")
+    with pytest.raises(Exception):
+        join_path("/test", "../..")
+
+
+def test_abs_and_rel_paths():
+    assert join_path('/', 'this/is/a/test/') == '/this/is/a/test'
+    assert join_path('.', 'this/is/a/test/') == 'this/is/a/test'
+    assert join_path('', 'this/is/a/test/') == 'this/is/a/test'
+    assert join_path('/test', '.') == '/test'
+    assert join_path('/test', '..', 'this') == '/this'
+    assert join_path('/test', '../this') == '/this'
```

Comment:
Some tests will improve confidence that `join_path()` works

Reply:
Thank you.